src/v/security/scram_credential.h

```diff
@@ -45,3 +46,41 @@ class scram_credential {
 std::ostream& operator<<(std::ostream&, const security::scram_credential&);
 
 } // namespace security
+
+// TODO: avoid bytes-to-iobuf conersion. either add bytes specialization to
+// reflection or wait for reflection-v2 which will have a new interface. in
+// either case, this is only used when managing users not on a hot path.
+namespace reflection {
+template<>
+struct adl<security::scram_credential> {
+    static constexpr int8_t current_version = 1;
+
+    void to(iobuf& out, security::scram_credential&& c) {
+        adl<int8_t>{}.to(out, current_version);
+        serialize(
+          out,
+          bytes_to_iobuf(c.salt()),
```

Comment:
i don't get what's the difference here between bytes and iobuf? why not serde-it to `bytes` directly?

Reply:
mm, I don't believe there is a bytes specialization in serde per the comment above.

Reply:
i think might be worth adding it to the serde types tho, should be the sstring in the reflection.h